backend/de.metas.manufacturing/src/main/java/org/eevolution/productioncandidate/async/GeneratePPOrderFromPPOrderCandidate.java

```diff
@@ -61,23 +80,19 @@ public Result processWorkPackage(@NonNull final I_C_Queue_WorkPackage workPackag
 		final Stream<I_PP_Order_Candidate> candidateStream = StreamSupport.stream(
 				Spliterators.spliteratorUnknownSize(orderCandidates, Spliterator.ORDERED), false);
 
-		final OrderGenerateResult result = ppOrderCandidateService.processCandidates(candidateStream, isDocComplete);
-
-		Loggables.addLog("Generated: {}", result);
-
-		return Result.SUCCESS;
+		return PPOrderCandidateProcessRequest.builder()
+				.isDocCompleteOverride(isDocCompleteOverride)
+				.autoProcessCandidatesAfterProduction(autoProcessCandidatesAfterProduction)
+				.sortedCandidates(getSortedCandidates(candidateStream))
+				.build();
 	}
 
-	@Nullable
-	private static Boolean getCompleteDocParamValue(@NonNull final IParams params)
+	@NonNull
+	private ImmutableList<PPOrderCandidateToAllocate> getSortedCandidates(@NonNull final Stream<I_PP_Order_Candidate> candidateStream)
 	{
-		final Object isDocComplete = params.getParameterAsObject(WP_COMPLETE_DOC_PARAM);
-
-		if (isDocComplete == null)
-		{
-			return null;
-		}
-
-		return StringUtils.toBoolean(isDocComplete);
+		return candidateStream.filter(orderCandidate -> !orderCandidate.isProcessed())
+				.map(PPOrderCandidateToAllocate::of)
+				.sorted(Comparator.comparing(PPOrderCandidateToAllocate::getHeaderAggregationKey))
+				.collect(ImmutableList.toImmutableList());
```

Comment:
We had very recently this PR: https://github.com/metasfresh/metasfresh/pull/13896

It's about  making sure that if a number of PP_Order_Candiadtes are enqueued together, then the order in which the PP_Orders are created (*and thus their order if you sort them by DocumentNo!*), is according to the PP_Order_Candidates' SeqNo.

Please make sure that (still) works :-)